### PR TITLE
fix(Scripts/SunwellPlateau): Sathrovarr's Curse of Boundless Agony

### PR DESCRIPTION
**Pull request by sogladev**
## Changes Proposed:

This PR proposes changes to:
-  [ ] Core (units, players, creatures, game systems).
-  [x] Scripts (bosses, spell scripts, creature scripts).
-  [x] Database (SAI, creatures, etc).

Add jump behavior to PLR spell (45034)

Add visual dummy spell to indicate power of curse

Prevent stacking of curses if already afflicted

Add dispel visual when the encounter completes

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/21674

## SOURCE:

The changes have been validated through:
- [ ] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [x] Sniffs (remember to share them with the open source community!)
- [x] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [ ] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks must be committed using the proper --author tag in order to be accepted, thus crediting the original authors, unless otherwise unable to be found**

## Tests Performed:

This PR has been:
- [x] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [ ] This pull request requires further testing and may have edge cases to be tested.

## How to Test the Changes:

- [x] This pull request can be tested by following the reproduction steps provided in the linked issue
- [ ] This pull request requires further testing. Provide steps to test your changes. If it requires any specific setup e.g multiple players please specify it as well.

1.
2.
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/scripts/EasternKingdoms/SunwellPlateau/boss_kalecgos.cpp [hunk 1]
```diff
@@ -476,8 +481,18 @@ class spell_kalecgos_curse_of_boundless_agony_aura : public AuraScript
...
             GetAura()->GetEffect(aurEff->GetEffIndex())->SetAmount(aurEff->GetAmount() * 2);
+
+        uint32 spellId = 0;
+        if (tickNumber <= 10)
+            spellId = SPELL_CURSE_OF_BOUNDLESS_AGONY_DUMMY_1;
+        else if (tickNumber <= 20)
+            spellId = SPELL_CURSE_OF_BOUNDLESS_AGONY_DUMMY_2;
+        else
+            spellId = SPELL_CURSE_OF_BOUNDLESS_AGONY_DUMMY_3;
+        GetTarget()->CastSpell(GetTarget(), spellId, true);
```
Comment (Kitzunu) [inline src/server/scripts/EasternKingdoms/SunwellPlateau/boss_kalecgos.cpp:495]: The target is supposed to cast it on itself?

**Review by Nyeriah — APPROVED** [review]

File: src/server/scripts/EasternKingdoms/SunwellPlateau/boss_kalecgos.cpp (hunk 1 above)
Comment (sogladev) [reply src/server/scripts/EasternKingdoms/SunwellPlateau/boss_kalecgos.cpp:495]: from spellwork
```
ID - 45083 Curse of Boundless Agony
=================================================
Category = 0, SpellIconID = 544, activeIconID = 0, SpellVisual = (10068,0), SpellPriority = 0
Family SPELLFAMILY_GENERIC, flag [0] 0x00000000 [1] 0x00000000 [2] 0x00000000

SpellSchoolMask = 32 (SPELL_SCHOOL_MASK_SHADOW)
DamageClass = 0 (SPELL_DAMAGE_CLASS_NONE)
PreventionType = 0 (SPELL_PREVENTION_TYPE_NONE)
=================================================
Attributes: 0x28000180 (SPELL_ATTR0_DO_NOT_DISPLAY, SPELL_ATTR0_DO_NOT_LOG, SPELL_ATTR0_ALLOW_WHILE_SITTING, SPELL_ATTR0_NO_IMMUNITIES)
AttributesEx2: 0x00000004 (SPELL_ATTR2_IGNORE_LINE_OF_SIGHT)
AttributesEx3: 0x20040100 (SPELL_ATTR3_ONLY_ON_PLAYER, SPELL_ATTR3_ALWAYS_HIT, SPELL_ATTR3_IGNORE_CASTER_MODIFIERS)
AttributesEx5: 0x00000008 (SPELL_ATTR5_ALLOW_WHILE_STUNNED)
=================================================
Spell Level = 67, base 67, max 0, maxTarget 0

Category = 0
DispelType = 0 (DISPEL_NONE)
Mechanic = 0 (MECHANIC_NONE)
SpellRange: (Id 1) "Self Only":
    MinRange = 0, MinRangeFriendly = 0
    MaxRange = 0, MaxRangeFriendly = 0

CastingTime (Id 1) = 0.00

Interrupt Flags: 0x00000000, AuraIF 0x00000000, ChannelIF 0x00000000
Chance = 101, charges - 0
=================================================
Effect 0: Id 3 (SPELL_EFFECT_DUMMY)
BasePoints = 1
Targets (1, 0) (TARGET_UNIT_CASTER, NO_TARGET)

Effect 1:  NO EFFECT

Effect 2:  NO EFFECT

```
